fsw/src/bpnode_dispatch.c

```diff
@@ -0,0 +1,135 @@
+    {
+        CFE_MSG_GetMsgId(MsgPtr, &MsgId);
+        CFE_MSG_GetFcnCode(MsgPtr, &FcnCode);
+
+        CFE_EVS_SendEvent(BPNODE_CMD_LEN_ERR_EID, CFE_EVS_EventType_ERROR,
+                          "Invalid Msg length: ID = 0x%X,  CC = %u, Len = %u, Expected = %u",
+                          (unsigned int)CFE_SB_MsgIdToValue(MsgId), (unsigned int)FcnCode, (unsigned int)ActualLength,
+                          (unsigned int)ExpectedLength);
+
+        result = false;
+
+        BPNode_AppData.ErrCounter++;
+    }
+
+    return result;
+}
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+/*                                                                            */
+/* BPNode ground command processing                                           */
+/*                                                                            */
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+void BPNode_ProcessGroundCommand(const CFE_SB_Buffer_t *SBBufPtr)
+{
+    CFE_MSG_FcnCode_t CommandCode = 0;
+
+    CFE_MSG_GetFcnCode(&SBBufPtr->Msg, &CommandCode);
+
+    /*
+    ** Process ground commands
+    */
+    switch (CommandCode)
+    {
+        case BPNODE_NOOP_CC:
+            if (BPNode_VerifyCmdLength(&SBBufPtr->Msg, sizeof(BPNode_NoopCmd_t)))
+            {
+                BPNode_NoopCmd((const BPNode_NoopCmd_t *)SBBufPtr);
+            }
+            break;
+
+        case BPNODE_RESET_COUNTERS_CC:
+            if (BPNode_VerifyCmdLength(&SBBufPtr->Msg, sizeof(BPNode_ResetCountersCmd_t)))
+            {
+                BPNode_ResetCountersCmd((const BPNode_ResetCountersCmd_t *)SBBufPtr);
+            }
+            break;
+
+        /* default case already found during FC vs length test */
+        default:
+            CFE_EVS_SendEvent(BPNODE_CC_ERR_EID, CFE_EVS_EventType_ERROR, "Invalid ground command code: CC = %d",
+                              CommandCode);
+            break;
+    }
+}
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+/*                                                                            */
+/*  Purpose:                                                                  */
+/*     This routine will process any packet that is received on the           */
+/*     command pipe.                                                          */
+/*                                                                            */
+/* * * * * * * * * * * * * * * * * * * * * * * *  * * * * * * *  * *  * * * * */
+void BPNode_TaskPipe(const CFE_SB_Buffer_t *SBBufPtr)
+{
+    CFE_SB_MsgId_t MsgId = CFE_SB_INVALID_MSG_ID;
+
+    CFE_MSG_GetMsgId(&SBBufPtr->Msg, &MsgId);
+
+    switch (CFE_SB_MsgIdToValue(MsgId))
+    {
+        case BPNODE_CMD_MID:
+            BPNode_ProcessGroundCommand(SBBufPtr);
+            break;
+
+        case BPNODE_SEND_HK_MID:
+            BPNode_SendHkCmd((const BPNode_SendHkCmd_t *)SBBufPtr);
+            break;
```

Comment:
The HK requests should be coming from the same MID as all other commands (just with a different function code).

Reply:
Will do

Reply:
Done